src/main/java/jenkins/bouncycastle/api/PEMManager.java

```diff
@@ -0,0 +1,400 @@
+     * <li>{@link org.bouncycastle.jce.provider.X509CertificateObject}
+     * <li>{@link org.bouncycastle.x509.X509V2AttributeCertificate}
+     * </ul>
+     * </ul>
+     * 
+     * @return {@link Object} read from the PEM
+     */
+    @CheckForNull
+    public Object getRawObject() {
+        return object;
+    }
+
+    /**
+     * Obtains the fingerprint of the private key in the "ab:cd:ef:...:12" format, which basically is an SHA1 digest
+     * from the key, encoded in hex format.
+     * 
+     * @return private key fingerprint in hex format "ab:cd:ef:...:12", null if the private key could not be obtained
+     * from the current PEM data.
+     * @throws IOException thrown if a problem exists creating the fingerprint
+     */
+    @CheckForNull
+    public String getPrivateKeyFingerprint() throws IOException {
+        PrivateKey key = toPrivateKey();
+        if (key == null) {
+            return null;
+        }
+        return hexEncode(getKeyDigestSHA1(key));
+    }
+
+    /**
+     * Obtains the fingerprint of the public key in the "ab:cd:ef:...:12" format, which basically is an MD5 digest from
+     * the key, encoded in hex format.
+     * 
+     * @return public key fingerprint in hex format "ab:cd:ef:...:12", null if the public key could not be obtained from
+     * the current PEM data.
+     * @throws IOException if a problem exists creating the fingerprint
+     */
+    @CheckForNull
+    public String getPublicKeyFingerprint() throws IOException {
+        PublicKey key = toPublicKey();
+        if (key == null) {
+            return null;
+        }
+        return hexEncode(getKeyDigestMD5(key));
+    }
+
+    /**
+     * Generates an SHA1 digest from a Key object
+     * 
+     * @param k the key to generate the digest from
+     * @return the generated digest
+     * @throws IOException if a problem exists creating the digest
+     * @see <a href="https://docs.oracle.com/javase/7/docs/api/java/security/MessageDigest.html"> JDK Digest
+     * Algorithms</a>
+     */
+    @Nonnull
+    public static byte[] getKeyDigestSHA1(@Nonnull Key k) {
+        try {
+            return getKeyDigest(k, "SHA1");
+        } catch (NoSuchAlgorithmException e) {
+            throw new AssertionError("SHA1 algorithm not found to create digest");
+        }
+    }
+
+    /**
+     * Generates an MD5 digest from a Key object
+     * 
+     * @param k the key to generate the digest from
+     * @return the generated digest
+     * @throws IOException if a problem exists creating the digest
+     * @see <a href="https://docs.oracle.com/javase/7/docs/api/java/security/MessageDigest.html"> JDK Digest
+     * Algorithms</a>
+     */
+    @Nonnull
+    public static byte[] getKeyDigestMD5(@Nonnull Key k) {
+        try {
+            return getKeyDigest(k, "MD5");
+        } catch (NoSuchAlgorithmException e) {
+            throw new AssertionError("MD5 algorithm not found to create digest");
```

Comment:
"MD5 algorithm support is mandated by Java Language Specification. See https://docs.oracle.com/javase/7/docs/api/java/security/MessageDigest.html"